logger/sdkLogger.go

```diff
@@ -0,0 +1,295 @@
+	return &SdkLogger{
+		loggingOptions: loggingOpt,
+		logger:         loggingOpt.logger,
+	}
+}
+
+// LogRequest request Logs an HTTP request.
+func (a *SdkLogger) LogRequest(request *http.Request) {
+	var logLevel = a.loggingOptions.level
+	var contentTypeHeader = a._getContentType(request.Header)
+	var url string
+	if a.loggingOptions.request.includeQueryInPath {
+		url = request.RequestURI
+	} else {
+		url = a._removeQueryParams(request.RequestURI)
+	}
+
+	a.logger.Log(
+		logLevel,
+		"Request %{method} %{url} %{contentType}",
+		map[string]any{
+			"method":      request.Method,
+			"url":         url,
+			"contentType": contentTypeHeader,
+		})
+
+	a._applyLogRequestOptions(logLevel, request)
+}
+
+// LogResponse Logs an HTTP response.
+func (a *SdkLogger) LogResponse(response *http.Response) {
+
+	level := a.loggingOptions.level
+	a.logger.Log(
+		level,
+		"Response %{statusCode} %{contentLength} %{contentType}",
+		map[string]any{
+			"statusCode":    response.StatusCode,
+			"contentLength": a._getContentLength(response.Header),
+			"contentType":   a._getContentType(response.Header),
+		},
+	)
+
+	a._applyLogResponseOptions(level, response)
+}
+
+func (a *SdkLogger) _applyLogRequestOptions(level Level, request *http.Request) {
+
+	logOp := a.loggingOptions.request
+	if logOp.headers {
+		a.logger.Log(level, "Request headers %{headers}",
+			map[string]any{"headers": a._extractHeadersToLog(
+				logOp.includeHeaders,
+				logOp.excludeHeaders,
+				logOp.whitelistHeaders,
+				request.Header,
+			)},
+		)
+	}
+
+	if logOp.body {
+		if bodyBytes, err := json.Marshal(request.Body); err == nil {
+			a.logger.Log(level, "Request body %{body}",
+				map[string]any{"body": string(bodyBytes)},
+			)
+		}
+
+	}
+}
+
+func (a *SdkLogger) _applyLogResponseOptions(level Level, response *http.Response) {
+
+	logOp := a.loggingOptions.response
+	if logOp.headers {
+		a.logger.Log(level, "Response headers %{headers}",
+			map[string]any{"headers": a._extractHeadersToLog(
+				logOp.includeHeaders,
+				logOp.excludeHeaders,
+				logOp.whitelistHeaders,
+				response.Header,
+			)},
+		)
+	}
+
+	if logOp.body {
+
+		if bodyBytes, err := json.Marshal(response.Body); err == nil {
+			a.logger.Log(level, "Response body %{body}",
+				map[string]any{"body": string(bodyBytes)},
+			)
+		}
+	}
+}
+
+const CONTENT_TYPE_HEADER = "content-type"
+const CONTENT_LENGTH_HEADER = "content-length"
+
+func (a *SdkLogger) _getContentType(headers http.Header) string {
+	var contentType string = ""
+	if len(headers) > 0 {
+		contentType = headers.Get(CONTENT_TYPE_HEADER)
+	}
+	return contentType
+}
+
+func (a *SdkLogger) _getContentLength(headers http.Header) string {
+	var contentLength string = ""
+	if len(headers) > 0 {
+		contentLength = headers.Get(CONTENT_LENGTH_HEADER)
+	}
+	return contentLength
+}
+
+func (a *SdkLogger) _removeQueryParams(url string) string {
+	if strIndex := strings.Index(url, "?"); strIndex != -1 {
+		return url[:strIndex]
```

Comment:
@haseeb-mhr You need to cover the uncovered lines in `sdkLogger.go` class

Reply:
updated request to improve code coverage.